analyses/chromosomal-instability/00-setup-breakpoint-data.R

```diff
@@ -215,14 +245,48 @@ sv_breaks <- data.frame(
     samples %in% common_samples
   )
 
-############################## Create union of breaks ##########################
-# Make an union of breaks data.frame.
-union_of_breaks <- dplyr::bind_rows(sv_breaks, cnv_breaks) %>%
-  dplyr::distinct(samples, chrom, coord, .keep_all = TRUE)
+# Remake this in case some samples' data got filtered out in the process
+common_samples <- dplyr::intersect(
+  unique(cnv_breaks$samples),
+  unique(sv_breaks$samples)
+)
+
+######################### Create intersection of breaks ########################
+# Make an intersection of breaks data.frame.
+intersection_of_breaks <- lapply(common_samples, function(sample_id) {
```

Comment:
I think I figured out why I don't love this. its because the function you define here uses sv_breaks and cnv_breaks, neither of which is an argument to the function itself. I'd rather see something like a function `intersect_breaks(sample_id, sv, cnv)` followed by `lapply(common_samples, intersect_breaks, sv = sv_breaks, cnv = cnv_breaks)`

Reply:
Made this its own specific argument outside of the lapply [in this commit. ](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/518/commits/31c81f816218d54ad9c1c1af3d77d42050c0bf68)